src/util/CMakeLists.txt

```diff
@@ -1,4 +1,8 @@
+file(GLOB util_HDRS "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
```

Comment:
why are we using glob for some cases but not all?

Reply:
Reason was 2 fold:

* GLOB is sometime discouraged or considered as bad practise https://discourse.cmake.org/t/is-glob-still-considered-harmful-with-configure-depends/808 , so I used it only where is was more needed
* but I mostly did this to show the 2 ways (listing vs glob), let reviewers discuss and agree on which is the best, and then later I update the PR with consistent method depending on what have been checked.

I don't have strong opinion on GLOB vs manual listing but I know some people may have

Reply:
Done. 
Just irr/src/CMakeLists.txt does not do it with GLOB because this folder contains different targets that don't all use all .h files. So this one makes more sense to manually select what belongs to what target.